packages/records/lib/models/records.ts

```diff
@@ -322,24 +313,27 @@ export async function update({
             for (let i = 0; i < recordsWithoutDuplicates.length; i += BATCH_SIZE) {
                 const chunk = recordsWithoutDuplicates.slice(i, i + BATCH_SIZE);
 
-                updatedKeys.push(...(await getUpdatedKeys({ records: chunk, connectionId, model, trx })));
+                const oldRecords = await getRecordsToUpdate({ records: chunk, connectionId, model, trx });
+                if (oldRecords.length === 0) {
+                    logger.warn(`Did not find any record to update: ${{ connectionId, model }}`);
+                }
 
                 const recordsToUpdate: FormattedRecord[] = [];
-                const rawOldRecords = await getRecordsByExternalIds({ externalIds: updatedKeys, connectionId, model, trx });
-                for (const rawOldRecord of rawOldRecords) {
-                    if (!rawOldRecord) {
+                for (const oldRecord of oldRecords) {
+                    const oldRecordData = decryptRecordData(oldRecord);
+
+                    const inputRecord = chunk.find((record) => record.external_id === oldRecord.external_id);
+                    if (!inputRecord) {
                         continue;
                     }
 
-                    const { record: oldRecord, ...oldRecordRest } = rawOldRecord;
-
-                    const record = records.find((record) => record.external_id === rawOldRecord.external_id);
+                    const { json: newRecordData, ...newRecordRest } = inputRecord;
 
                     const newRecord: FormattedRecord = {
-                        ...oldRecordRest,
+                        ...newRecordRest,
```

Comment:
that was a bug. The only attributes that is different is the sync_job_id and was not updated